src/GregClient/Requests/BanPackage.cs

```diff
@@ -0,0 +1,37 @@
+﻿using RestSharp;
+
+namespace Greg.Requests
+{
+    public class BanPackage : Request
+    {
+        private enum Action { Ban, Unban }
+        private readonly Action banAction;
+        private readonly string packageId;
+
+        public BanPackage(string packageId, bool banPackage)
+        {
+            // both endpoints require authentication by the user
+            this.ForceAuthentication = true;
```

Comment:
By default, `PUT` and `POST` requests require authentication.  So, I don't believe you need this.